### ISSUE TITLE
Some failed downloads fail to resume

### ISSUE THREAD

**Issue by Lesik** [tranxuanthang/villain3ds#10]
If a partial content download has failed five times, but is manually resumed by clicking the play button, it sometimes fails with

```
Content #00000010 is failed to download (Error: Invalid file path: /home/me/Villain3DS/raw/0004008c000ba900/00000010.mtd).
```

By clicking the play button again, the message changes to

```
Content #00000010 is already downloaded before.
```

Sometimes the download resumes correctly without problems.

I doubt this is intended behavior. This is especially annoying for titles with many partial contents, such as Pokémon™ Mystery Dungeon: Gates to Infinity.

**Comment by Lesik** [comment]
[Here's a screengrab](https://thumbs.gfycat.com/OrnateFaintEyas-mobile.mp4) showing one partial content that successfully resumes download, one that fails, and another one that works.

**Comment by tranxuanthang** [comment]
Sorry for my late reply my friend.

At first I'll do something about titles which have many small contents, like dlc. Let it download one-by-one might reduce request timed out error.

And now the "invalid file path" error part. This is what I'm working on.
I can reproduce this error by download a random title which is <100MB, let it download until 95% and click pause button. Then after clicking "play" button to resume the download, I'll sometimes get the "invalid file path" error.
It could be because of the file is already completed downloading. By the time the content is paused, it completes the download and it is renamed from 0000xxxx.mtd to 0000xxxx. I tested this multiple time and the md5/sha1 hash was matched (most of the time, in some cases the file is actually corrupted). The progression we see on our screen is still show 95% somehow, maybe because it updates slower than the real stats.
Dlc titles have multiple small contents (just around one-hundred kilobytes in size) so this bug can be seen more frequently.

And the last thing is make_cdn_cia file is somehow always crashed (stopped working) on both windows and linux system while trying to build cia file for dlcs (or titles with many many contents). I tested with funkeycia and it has the same problem (but usb helper is still working fine).
